lexicon/conversion.py

```diff
@@ -472,12 +484,23 @@ def run(self):
                         if eval not in target_lemma.eval:
                             target_lemma.eval.append(eval)
                     if source_lemma in lex.lemmata:
-                        lex.lemmata.remove(source_lemma)
+                        if not self.keep_original_target_lemmas:
+                            lex.lemmata.remove(source_lemma)
+                        else:
+                            # Replace the source lemma and keep original target lemma as well
+                            # without changing the position in the lexicon
+                            source_position = lex.lemmata.index(source_lemma)
+                            target_position = lex.lemmata.index(target_lemma)
+                            lex.lemmata[source_position] = target_lemma
+                            lex.lemmata[target_position] = copy_target_lemma
                 if not target_lemma.synt:
                     if source_lemma and source_lemma.synt:
                         target_lemma.synt = source_lemma.synt
                     else:
                         target_lemma.synt = source_orth.split()
+                        if self.keep_original_target_lemmas:
```

Comment:
`copy_target_lemma` could be inserted twice for the case `source_lemma` exists but `source_lemma.synt` not.

Reply:
I changed it now that its only added if source_lemma is not there.